accord-core/src/test/java/accord/utils/Gens.java

```diff
@@ -134,6 +157,50 @@ public Gen<Boolean> all()
         {
             return RandomSource::nextBoolean;
         }
+
+        public Gen<Boolean> runs(double ratio)
+        {
+            Invariants.checkArgument(ratio > 0 && ratio <= 1, "Expected %d to be larger than 0 and <= 1", ratio);
+            int steps = (int) (1 / ratio);
+            double lower = ratio * .8;
+            double upper = ratio * 1.2;
+            return new Gen<Boolean>() {
+                private int run = -1;
+                private long falseCount = 0, trueCount = 0;
+                @Override
+                public Boolean next(RandomSource rs)
+                {
+                    if (run != -1)
```

Comment:
Is the goal of this to generate a `step` number of true values every time an unlikely value is created?

Maybe add a comment explaining what this can be used to do.

Reply:
added the following

```
// run represents how many consecutaive true values should be returned; -1 implies no active "run" exists
```

let me know if this helps.

> Is the goal of this to generate a step number of true values every time an unlikely value is created?


I wouldn't say "unlikely value is created".  This function tries to return `true` within the given `ratio` but adds a bias that multiple `true`s can and will happen.  This was pulled from https://issues.apache.org/jira/browse/CASSANDRA-18451 which uses this to simulation partitions and ephemeral issues... the idea in those cases are that when an error (such as partition) happens, it will happen for a few samples then resolve... so `run` represents these samples